pkg/openshift/rosa/cluster.go

```diff
@@ -306,19 +306,19 @@ func (r *Provider) validateCreateClusterOptions(options *CreateClusterOptions) (
 		}
 	}
 
-	if options.accountRoles.controlPlaneRoleARN == "" {
+	if options.accountRoles.controlPlaneRoleARN == "" && (options.HostedCP || options.STS) {
 		errs = append(errs, errors.New("iam role arn for control plane is required"))
 	}
 
-	if options.accountRoles.installerRoleARN == "" {
+	if options.accountRoles.installerRoleARN == "" && (options.HostedCP || options.STS) {
 		errs = append(errs, errors.New("iam role arn for installer is required"))
 	}
 
-	if options.accountRoles.supportRoleARN == "" {
+	if options.accountRoles.supportRoleARN == "" && (options.HostedCP || options.STS) {
 		errs = append(errs, errors.New("iam role arn for support role is required"))
 	}
 
-	if options.accountRoles.workerRoleARN == "" {
+	if options.accountRoles.workerRoleARN == "" && (options.HostedCP || options.STS) {
 		errs = append(errs, errors.New("iam role for worker role is required"))
 	}
```

Comment:
Can we gate all of these arn checks with a single `if options.HostedCP || options.STS { ...` instead of repeating the condition?